server/api/chat/chat.controllers.js

```diff
@@ -83,16 +84,36 @@ export const getChatByNames = asyncHandler(async (req, res, next) => {
   let { messagesHistory } = userChat;
   delete userChat._id;
   messagesHistory.sort((a, b) => a.createdAt - b.createdAt);
-  const messagesTimeToLocal = messagesHistory.map((message) => {
+  messagesHistory.forEach((message) => {
     delete message._id;
-    return {
-      ...message,
-      createdAt: message.createdAt.toLocaleString("en-US", {
-        timeZone: "Asia/Jerusalem",
-        hour12: false,
-      }),
-    };
+    return message;
   });
-  userChat.messagesHistory = messagesTimeToLocal;
   res.status(200).json(userChat);
 });
+
+//$ @desc    get user Chats list with names and last message
+//$ @route   GET /api/chat/logged/user/:loggedUser_name
+//! @access  NOT SET YET
+export const getUserChatsList = asyncHandler(async (req, res, next) => {
+  const { loggedUser_name: name } = req.params;
+
+  let userChats = await Chat.find({
+    users: { $all: [name] },
+  })
+    .sort({ updatedAt: -1 })
+    .lean();
+  if (!userChats) return next(new Error("failed retrieving"));
+  if (userChats.length > 0) {
+    userChats = userChats.map((chat) => {
+      const { users, messagesHistory } = chat;
+      chat.recieverName = users.filter((user) => user !== name)[0];
```

Comment:
what is the point of attaching in to the "chat" object? you return @ line 112 only this specific field anyway.
just create a new var and return it